web/main.go

```diff
@@ -0,0 +1,344 @@
+		},
+	}
+}
+
+// Define the template registry struct
+type TemplateRegistry struct {
+	templates map[string]*template.Template
+}
+
+// Implement e.Renderer interface
+func (t *TemplateRegistry) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
+	tmpl, ok := t.templates[name]
+	if !ok {
+		err := errors.New("Template not found -> " + name)
+		return err
+	}
+	return tmpl.ExecuteTemplate(w, "layout.html", data)
+}
+
+func getCandidatos(filters *HomeFilters, _offset int) *[]Candidate {
+	if filters.State == "" || filters.City == "" {
+		return &[]Candidate{}
+	}
+
+	return &[]Candidate{
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+		newCandidate(),
+	}
+}
+
+func getTeamMembers() []TeamMember {
+	return []TeamMember{
+		newTeamMember("Ana Paula Gomes", "Pitaqueira", "/img/team/ana.jpeg", []SocialLink{
+			newSocialLink("linkedin", "#"),
+		}),
+		newTeamMember("Aurélio Buarque", "Desenvolvedor", "/img/team/aurelio.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("github", "#"),
+		}),
+		newTeamMember("Bruno Morassutti", "Palpiteiro jurídico", "/img/team/bruno.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("linkedin", "#"),
+		}),
+		newTeamMember("Daniel Fireman", "Coordenador", "/img/team/daniel.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("github", "#"),
+			newSocialLink("linkedin", "#"),
+			newSocialLink("instagram", "#"),
+		}),
+		newTeamMember("Eduardo Cuducos", "Palpiteiro", "/img/team/eduardo.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("github", "#"),
+		}),
+		newTeamMember("Evelyn Gomes", "Articuladora ", "/img/team/evelyn.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("linkedin", "#"),
+			newSocialLink("github", "#"),
+		}),
+		newTeamMember("Laura Cavalcante", "Advogada ", "/img/team/laura.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("linkedin", "#"),
+		}),
+		newTeamMember("Mariana Souto", "Designer ", "/img/team/mariana.jpeg", []SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("instagram", "#"),
+			newSocialLink("linkedin", "#"),
+		}),
+	}
+}
+
+func findCandidate(_token string) Candidate {
+	return newCandidate()
+}
+
+func homeHandler(c echo.Context) error {
+	year := c.QueryParam("ano")
+	if year == "" {
+		year = strconv.Itoa(time.Now().Year())
+	}
+	state := c.QueryParam("estado")
+	citiesByState := map[string]interface{}{
+		"Alagoas": [2]string{"Maceió", "Arapiraca"},
+		"Bahia":   [1]string{"Salvador"},
+	}
+	cities, err := citiesByState[state]
+	if !err {
+		cities = []string{}
+	}
+
+	filters := newHomeFilters(state, year, c.QueryParam("cidade"), c.QueryParam("cargo"))
+	offset, _ := strconv.Atoi(c.QueryParam("offset"))
```

Comment:
Tratar o erro que a função Atoi retorna:
```suggestion
	offset, err := strconv.Atoi(c.QueryParam("offset"))
	if err != nil {
	     return fmt.Error("failed to parse offset string [%s] to int, error %v", c.QueryParam("offset"), err)
	}
```